### PR TITLE
Validating transaction isolation level for unsupported values during configuration

### LINKED ISSUE #132: MSMQ catch invalid TransactionScope configurations at startup
based on Particular/NServiceBus#4237, we can detect unsupported/invalid configurations at startup and prevent "runtime" errors  (during message handling). Instead we can throw on endpoint startup which makes it easier to detect the issue.

Unsupported configurations:
- when using TransportTransactionMode.TransactionScope
  - TransactionScopeOptions(isolationLevel:IsolationLevel.Snapshot)
- when using a different TransportTransactionMode, any TransactionScopeOptions usage should be avoided.

any other invalid transaction configurations (e.g. invalid timeout settings)?

### PR DESCRIPTION
**Pull request by ramonsmits**
Fixes #132

### CODE REVIEW

File: src/NServiceBus.Transport.Msmq/MsmqConfigurationExtensions.cs [hunk 1]
```diff
@@ -43,10 +43,16 @@ public static TransportExtensions<MsmqTransport> TransactionScopeOptions(this Tr
         {
             Guard.AgainstNull(nameof(transportExtensions), transportExtensions);
             Guard.AgainstNegativeAndZero(nameof(timeout), timeout);
+
+            if (isolationLevel.HasValue && isolationLevel.Value == IsolationLevel.Snapshot)
+            {
+                throw new ArgumentException("Isolation level `Snapshot` is unsupport by the transport. Consider not sharing the transaction between transport and persistence if persistence should use `IsolationLevel.Snapshot` by using `TransportTransactionMode.SendsAtomicWithReceive` or lower.", nameof(isolationLevel));
```
Comment (mikeminutillo) [inline src/NServiceBus.Transport.Msmq/MsmqConfigurationExtensions.cs:49]:
```suggestion
                throw new ArgumentException("Isolation level `Snapshot` is unsupported by the transport. Consider not sharing the transaction between transport and persistence if persistence should use `IsolationLevel.Snapshot` by using `TransportTransactionMode.SendsAtomicWithReceive` or lower.", nameof(isolationLevel));
```

File: src/NServiceBus.Transport.Msmq.AcceptanceTests/When_customizing_scope_isolation_level.cs [hunk 2]
```diff
@@ -34,9 +71,9 @@ public ScopeEndpoint()
             {
                 EndpointSetup<DefaultServer>(c =>
                 {
-                    c.UseTransport<MsmqTransport>()
-                        .Transactions(TransportTransactionMode.TransactionScope)
-                        .TransactionScopeOptions(isolationLevel: IsolationLevel.RepeatableRead);
+                    c.UseTransport<MsmqTransport>();
+                    //.Transactions(TransportTransactionMode.TransactionScope)
```
Comment (kbaley) [inline src/NServiceBus.Transport.Msmq.AcceptanceTests/When_customizing_scope_isolation_level.cs:75]: Did you mean to leave these in?

File: src/NServiceBus.Transport.Msmq.AcceptanceTests/When_customizing_scope_isolation_level.cs [hunk 3]
```diff
@@ -10,17 +11,53 @@
...
+                                    .Transactions(TransportTransactionMode.TransactionScope)
+                                    .TransactionScopeOptions(isolationLevel: IsolationLevel.Snapshot);
+                            });
+                            g.When(b => b.SendLocal(new MyMessage()));
+                        })
+                        .Done(c => c.Done)
+                        .Run();
+            });
+
+            StringAssert.Contains("Isolation level `Snapshot` is unsupport by the transport. Consider not sharing the transaction between transport and persistence if persistence should use `IsolationLevel.Snapshot` by using `TransportTransactionMode.SendsAtomicWithReceive` or lower.", ex.Message);
```
Comment (kbaley) [inline src/NServiceBus.Transport.Msmq.AcceptanceTests/When_customizing_scope_isolation_level.cs:59]:
```suggestion
            StringAssert.Contains("Isolation level `Snapshot` is not supported by the transport. Consider not sharing the transaction between transport and persistence if persistence should use `IsolationLevel.Snapshot` by using `TransportTransactionMode.SendsAtomicWithReceive` or lower.", ex.Message);
```

File: src/NServiceBus.Transport.Msmq.AcceptanceTests/When_customizing_scope_isolation_level.cs (hunk 2 above)
Comment (ramonsmits) [reply src/NServiceBus.Transport.Msmq.AcceptanceTests/When_customizing_scope_isolation_level.cs]: @kbaley nope, cleaned it.

**Pull request merged**